### PR TITLE
Feature/peer 643

### PR DESCRIPTION
**Pull request by YuriyLu**

### CODE REVIEW

File: app/containers/EthereumProvider/index.js [hunk 1]
```diff
@@ -144,19 +144,16 @@ export const EthereumProvider = ({
...
+  useEffect(() => {
+    if (ethereum) {
+      ethereum.setData({
+        wallet,
+        connectedWallets,
+        web3Onboard,
+        connectedChain,
+      });
+    }
+  }, [wallet, connectedWallets, web3Onboard]);
```
Comment (Lampardks) [inline app/containers/EthereumProvider/index.js:156]: add ethereum

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+const TransactionsList: React.FC<TransactionsListProps> = ({
+  transactionList,
+  transactionInPending,
+}): JSX.Element => {
+  const [opened, setOpened] = useState(false);
+
+  useEffect(() => {
+    const transactionsStatusBlock = document.getElementById(
+      'transactions-status-block',
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:43]: use ref

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    );
+    if (!transactionList.length && !transactionInPending) {
+      transactionsStatusBlock.style.right = '-289px';
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:46]: use state

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  const successfulTransactions = transactionList.filter(
+    (transaction) => transaction.result?.status === 1,
+  );
+  const transactionsInProgress = transactionList.filter(
+    (transaction) => transaction.result?.status !== 1,
+  );
+
+  useEffect(() => {
+    if (transactionList.length && transactionInPending && !opened) {
+      document.getElementById('status-button').style.width = '449px';
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:64]: use ref

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      setTimeout(
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:65]: add clear timeOut

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+  return (
+    <div id="transactions-status-block" css={styles.transactionsStatusBlock}>
+      {opened && <div css={styles.blocker} />}
+
+      <div
+        id="status-button"
+        className="df fdr aic pl16 pa"
+        css={styles.statusButton}
+        onClick={() => {
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:80]: move it to a separate function

File: app/containers/TransactionsList/TransactionsList.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+                <FormattedMessage id={commonMessages.successful.id} />
+              ) : (
+                <FormattedMessage id={commonMessages.failed.id} />
+              );
+          } else {
+            status = <FormattedMessage id={commonMessages.inProgress.id} />;
+          }
+
+          return (
+            <SingleTransaction
```
Comment (Lampardks) [inline app/containers/TransactionsList/TransactionsList.tsx:149]: move it to a separate component

File: app/utils/ethereum.js [hunk 8]
```diff
@@ -246,9 +242,29 @@ class EthereumService {
...
+      ).result = result;
+      setTimeout(() => {
+        const index = this.transactionList
+          .map((transactionFromList) => transactionFromList.transactionHash)
+          .indexOf(transaction.hash);
+        if (index !== -1) {
+          this.transactionList.splice(index, 1);
+          this.setTransactionList(this.transactionList);
+        }
+      }, '30000');
```
Comment (Lampardks) [inline app/utils/ethereum.js:265]: add clear timeOut

File: app/utils/ethereum.js [hunk 9]
```diff
@@ -299,7 +315,15 @@ class EthereumService {
   ) => {
     await this.chainCheck();
     const metaTxContract = this[contract];
-    const nonce = await metaTxContract.getNonce(actor);
+    let nonce = await metaTxContract.getNonce(actor); //упорядочивает
```
Comment (Lampardks) [inline app/utils/ethereum.js:318]: delete ru comment

File: app/utils/ethereum.js [hunk 10]
```diff
@@ -357,16 +381,38 @@ class EthereumService {
...
+    ).result = result;
+    setTimeout(() => {
+      const index = this.transactionList
+        .map((transactionFromList) => transactionFromList.transactionHash)
+        .indexOf(response.body.transactionHash);
+      if (index !== -1) {
+        this.transactionList.splice(index, 1);
+        this.setTransactionList(this.transactionList);
+      }
+    }, '30000');
```
Comment (Lampardks) [inline app/utils/ethereum.js:413]: add clear timeOut

File: app/containers/TransactionsList/TransactionsList.tsx (hunk 2 above)
Comment (YuriyLu) [reply app/containers/TransactionsList/TransactionsList.tsx:43]: done

File: app/containers/TransactionsList/TransactionsList.tsx (hunk 3 above)
Comment (YuriyLu) [reply app/containers/TransactionsList/TransactionsList.tsx:46]: done

File: app/containers/TransactionsList/TransactionsList.tsx (hunk 5 above)
Comment (YuriyLu) [reply app/containers/TransactionsList/TransactionsList.tsx:65]: done

File: app/containers/TransactionsList/TransactionsList.tsx (hunk 6 above)
Comment (YuriyLu) [reply app/containers/TransactionsList/TransactionsList.tsx:80]: done

File: app/utils/ethereum.js (hunk 9 above)
Comment (YuriyLu) [reply app/utils/ethereum.js:318]: done

File: app/containers/TransactionsList/TransactionsList.tsx (hunk 7 above)
Comment (YuriyLu) [reply app/containers/TransactionsList/TransactionsList.tsx:149]: simplified

**Review by Lampardks — APPROVED** [review]

**Pull request merged**